### PR TITLE
add support for openrc

### LINKED ISSUE #266: can you add support for openrc?

### PR DESCRIPTION
**Pull request by tomjelinek**
resolves #266

### CODE REVIEW

File: pcs_test/tier0/common/services/drivers/test_openrc.py [hunk 1]
```diff
@@ -0,0 +1,256 @@
...
+        )
+
+
+class IsInstalledTest(Base):
+    def test_installed(self):
+        output = (
+            "service1       	0:off	1:off	2:on	3:on	4:on	5:on	6:off\n"
+            "abc            	0:off	1:on	2:on	3:on	4:on	5:on	6:off\n"
+            "xyz            	0:off	1:off	2:off	3:off	4:off	5:off	6:off\n"
+            f"{self.service}        	0:off	1:off	2:on	3:on	4:on	5:on	6:off\n"
```
Comment (Alessandro-Barbieri) [inline pcs_test/tier0/common/services/drivers/test_openrc.py:208]: tests will surely fail
see
https://github.com/gentoo/guru/blob/master/sys-cluster/pcs/files/pcs-0.11-gentoo-support.patch

File: pcs/common/services/drivers/openrc.py [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+            service = service.split(" ", 1)[0]
+            if service:
+                service_list.append(service)
+        return service_list
+
+    def is_current_system_supported(self) -> bool:
+        return all(
+            os.path.isfile(binary)
+            for binary in (self._rc_service_bin, self._rc_config_bin)
+        )
```
Comment (ondrejmular) [inline pcs/common/services/drivers/openrc.py:98]: This file is missing in `pcs/Makefile.am`

**Review by ondrejmular — CHANGES_REQUESTED** [review]

File: pcs/common/services/drivers/openrc.py (hunk 2 above)
Comment (tomjelinek) [reply pcs/common/services/drivers/openrc.py:98]: fixed

**Comment by custom82** [comment]
checking for GNU default python exec_prefix... ${exec_prefix}
checking for python script directory (pythondir)... ${PYTHON_PREFIX}/lib/python3.9/site-packages
checking for python extension module directory (pyexecdir)... ${PYTHON_EXEC_PREFIX}/lib/python3.9/site-packages
checking for systemd... no
configure: error: Package requirements (systemd) were not met:

Package 'systemd', required by 'virtual:world', not found

Consider adjusting the PKG_CONFIG_PATH environment variable if you
installed software in a non-standard prefix.

Alternatively, you may set the environment variables systemd_CFLAGS
and systemd_LIBS to avoid the need to call pkg-config.
See the pkg-config man page for more details.

**Comment by tomjelinek** [comment]
@custom82 So there is this nice patch https://github.com/gentoo/guru/blob/master/sys-cluster/pcs/files/pcs-0.11-gentoo-support.patch fixing probably all the OpenRC and Gentoo related issues. Why don't you take it a bit further and submit it as a pull request?

**Comment by custom82** [comment]
I really trusted a lot in your patch, our patch written by me and Alessandro Barbieri we don't know how functional it is.

**Comment by tomjelinek** [comment]
@custom82

Well, we have virtually no knowledge of and experience with Gentoo. So if you are able to patch pcs to make it work on Gentoo (which according to the patch you are) and you have access to Gentoo machines to test it, then you already have a big advantage.

Not only you wrote openrc driver including tests, you also fixed pam config. Your patch is a great start and only needs a bit of polishing. Mainly, make systemd detection optional instead of removing it. Also, it would be great if replacing `localstatedir` variable with `/var` could somehow be resolved. Isn't `localstatedir` equal to `/var` on Gentoo?

It would really make it easier for us if you submitted you patch in a pull request.

**Comment by tomjelinek** [comment]
Open pull requests consume resources in our CI systems. Since there was no response from the author of this PR for more than 18 months, I am closing this PR. If you decide to implement requested changes and get your code ready to merge, feel free to open a new pull request and we'll be happy to review it.

**Pull request closed without merging**